sklearn/tests/test_common.py

```diff
@@ -268,3 +275,40 @@ def test_strict_mode_check_estimator():
 def test_strict_mode_parametrize_with_checks(estimator, check):
     # Ideally we should assert that the strict checks are Xfailed...
     check(estimator)
+
+
+def _generate_search_cv_instances():
```

Comment:
Can't we just have this inside of `construct_instance`, and rely on `test_estimators()` instead of having a new `test_search_cv()`?

Reply:
It was the alternative proposed in https://github.com/scikit-learn/scikit-learn/pull/18047 with concerns raised by @jnothman in the way that what we are creating is not generic for usual estimator (and third-party library).

In addition, this solution avoids returning a tuple of instances which is also a big change